src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs

```diff
@@ -11,6 +11,9 @@ public class PipeOptions
     {
         private const int DefaultMinimumSegmentSize = 4096;
 
+        // Arbitrary 1MB max segment size
+        internal const int MaximumSegmentSize = 1024 * 1024;
```

Comment:
Maybe this should be higher for large upload scenarios? Not sure.

Reply:
We should think about how this interacts with logic like that for [SocketConnection.MinAllocBufferSize](https://github.com/dotnet/aspnetcore/blob/57c3104f0d19bb9e25b8ba75b422e2d6072e1dd5/src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs#L15). I could see us potentially unnecessarily doing syscalls using relatively small 2KB buffer for reads after this change as we reach the end of the block. This is a DoS mitigation given smaller segment sizes. But with larger segment sizes, we could leave more bytes unfilled before allocating syscalls without wasting too much memory proportionally.

Reply:
Right, we'll need to tweak how we think about this (and if it matters). You'll end up throwing away 2K (which maybe is too aggressive anyways?)

Reply:
Are you suggesting anything or just saying this is interesting?